colossalai/fx/profiler/dataflow.py

```diff
@@ -0,0 +1,133 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Dict
+from torch.fx import Graph, Node
+from .memory import activation_size
+
+
+class Stage(Enum):
+    F = 0
+    L = 1
+    B = 2
+    P = 3
+
+
+@dataclass
+class GraphInfo:
+    """
+    GraphInfo is a dataclass for the dataflow analysis.
+    The dataflow analysis is conducted on a single node of the FX graph.
+    ============================================================================
+                            -------------------------------
+                            |            Node             |
+    [fwd_in] are       ---> | [fwd_in]          [bwd_out] |    <----- [bwd_out] is marks the memory for `grad_out`
+    placeholders saved for  |     | \__________     |     |
+    backward.               |     |            \    |     |
+                            | [fwd_tmp] ------> [bwd_tmp] |    <-----
+                            |     |  \_________     |     |    [bwd_tmp] marks the peak memory 
+                            |    / \           \    |     |    in backward pass.
+    [x] is not counted ---> | [x]  [fwd_tmp] -> [bwd_tmp] |    <-----
+    in [fwd_tmp] because    |  |       |  \_____    |     |
+    it is not saved for     |  |       |        \   |     |
+    backward.               -------------------------------
+    ============================================================================
+    Attributes:
+        fwd_in (int): See the above illustration.
+        fwd_tmp (int): See the above illustration.
+        bwd_tmp (int): See the above illustration.
+        bwd_out (int): See the above illustration.
+    """
+    fwd_in: int = 0
+    fwd_tmp: int = 0
+    bwd_tmp: int = 0
+    bwd_out: int = 0
```

Comment:
This class seems quite redundant with MetaInfo, you should use only one class. You set set flops to None by default.  Meanwhile, just for clarity, it is better to name the attribute with `mem` such as `fwd_in_mem`. In this way, I know `fwd_flop` is for computation and `fwd_in_mem` is for memory. Otherwise, I have no idea what is `fwd_in`.

Reply:
I see